framework/base/Controller.php

```diff
@@ -221,6 +221,13 @@ public function createAction($id)
         }
 
         $actionMap = $this->actions();
+        $this->ensureBehaviors();
+        foreach ($this->getBehaviors() as $behavior) {
+            if ($behavior instanceof ControllerBehaviorInterface) {
+                $actionMap = array_replace_recursive($actionMap, $behavior->actions());
```

Comment:
Could also add a second parameter to control what happens if the key already exists to make debugging less ambiguous:

```php
$array1 = [
        'error' => 'yii\web\ErrorAction',
        'view' =>  'yii\web\ViewAction',
        'custom' => 'my\custom\CustomAction',
 ];
$array2 = [
        'error' => 'yii\web\ErrorAction',
        'view' => new \yii\helpers\ArrayValueIfKeyNotExists(  'my\custom\ViewAction' , function() { 
                               throw new Exception('View action already defined.'); 
        }),
        'custom' => new \yii\helpers\ArrayValueIfKeyNotExists(  'my\custom\CustomAction'),
 ];

// would throw an exception
$result = ArrayHelper::merge($array1, $array2);

```